src/composables/useFeatureFlags.ts

```diff
@@ -1,40 +1,77 @@
-import { computed, reactive, readonly } from 'vue'
+import { computed, readonly } from 'vue'
 
-import { api } from '@/scripts/api'
+import { useFeatureFlagsStore } from '@/stores/featureFlagsStore'
 
 /**
- * Known server feature flags (top-level, not extensions)
+ * Known server feature flags
  */
 export enum ServerFeatureFlag {
+  // Core features
   SUPPORTS_PREVIEW_METADATA = 'supports_preview_metadata',
   MAX_UPLOAD_SIZE = 'max_upload_size',
+
+  // Extension features
   MANAGER_SUPPORTS_V4 = 'extension.manager.supports_v4'
 }
 
+/**
+ * Known client feature flags
+ */
+export enum ClientFeatureFlag {
+  SUPPORTS_MANAGER_V4_UI = 'supports_manager_v4_ui'
+}
+
 /**
  * Composable for reactive access to feature flags
  */
 export function useFeatureFlags() {
-  // Create reactive state that tracks server feature flags
-  const flags = reactive({
-    get supportsPreviewMetadata() {
-      return api.getServerFeature(ServerFeatureFlag.SUPPORTS_PREVIEW_METADATA)
-    },
-    get maxUploadSize() {
-      return api.getServerFeature(ServerFeatureFlag.MAX_UPLOAD_SIZE)
-    },
-    get supportsManagerV4() {
-      return api.getServerFeature(ServerFeatureFlag.MANAGER_SUPPORTS_V4)
-    }
-  })
+  const featureFlagsStore = useFeatureFlagsStore()
+
+  // Expose commonly used flags directly as computed properties
+  const flags = computed(() => ({
+    // Server flags
+    supportsPreviewMetadata: featureFlagsStore.getServerFeature<boolean>(
+      ServerFeatureFlag.SUPPORTS_PREVIEW_METADATA,
+      false
+    ),
+    maxUploadSize: featureFlagsStore.getServerFeature<number>(
+      ServerFeatureFlag.MAX_UPLOAD_SIZE
+    ),
+    supportsManagerV4: featureFlagsStore.supportsManagerV4,
+
+    // Client flags
+    clientSupportsManagerV4UI: featureFlagsStore.clientSupportsManagerV4UI,
+
+    // Store ready state
+    isReady: featureFlagsStore.isReady
```

Comment:
This feels like creating two ways of doing the same thing.
When would a developer choose to do

```ts
const { supportsManagerV4 } = useFeatureFlags()
```

over

```ts
const { supportsManagerV4 } = useFeatureFlagsStore()
```